webapp/src/components/modals/audio_group_call/audio_group_call.jsx

```diff
@@ -0,0 +1,427 @@
+
+import {initializeMyAudio, audioConnectToSwarm} from '../../../actions';
+
+import {id as pluginId} from 'manifest';
+
+async function getMediaStream(opts) {
+    return navigator.mediaDevices.getUserMedia(opts);
+}
+
+async function getMyStream() {
+    const audio = {
+        autoGainControl: true,
+        sampleRate: {ideal: 48000, min: 35000},
+        echoCancellation: true,
+        channelCount: {ideal: 1},
+        volume: 1,
+    };
+
+    try {
+        console.log('try just audio');
+
+        // If that fails, try just audio
+        const stream = await getMediaStream({audio});
+        return {myStream: stream, audioEnabled: true, videoEnabled: false};
+    } catch (err) {
+        console.error(err);
+
+        return {myStream: null, audioEnabled: false, videoEnabled: false};
+    }
+}
+
+class AudioCallPanel extends React.Component {
+    static propTypes = {
+        // eslint-disable-next-line react/no-unused-prop-types
+        getCurrentUser: PropTypes.func.isRequired,
+        userId: PropTypes.string.isRequired,
+    };
+    constructor(props) {
+        super(props);
+
+        const roomName = '2ecbcc63-0305-4edc-a351-1123913ba43f';
+        const invalidRoom = false;
+
+        const {configLoaded,
+            signalhubURL,
+            stunServer,
+            turnServer,
+            turnServerUsername,
+            turnServerCredential} = props;
+
+        this.state = {
+            initialized: false,
+            roomName,
+            invalidRoom,
+            peerStreams: {},
+            playBacks: {},
+            swarmInitialized: false,
+            myUuid: props.userId,
+            audioOn: false,
+            videoOn: false,
+            audioEnabled: true,
+            videoEnabled: false,
+            userId: props.userId,
+            speakerOn: false,
+            myUsername: props.username,
+            configLoaded,
+            signalhubURL,
+            stunServer,
+            turnServer,
+            turnServerUsername,
+            turnServerCredential,
+        };
+
+        console.log('PROPS', props, this.state);
+    }
+
+    async handleRequestPerms() {
+        const {myStream, audioEnabled, videoEnabled} = await getMyStream();
+        console.log({audioEnabled, videoEnabled});
+        this.setState({initialized: true, myStream, audioEnabled, videoEnabled});
+    }
+
+    async handleSetUserId(userId) {
+        this.setState({
+            userId,
+        });
+
+        this.connectToSwarm(userId);
+    }
+
+    connectToSwarm(userId) {
+        const {myUuid, myUsername, configLoaded,
+            signalhubURL,
+            stunServer,
+            turnServer,
+            turnServerUsername,
+            turnServerCredential} = this.state;
+        const roomCode = '6e0731b0-f185-45e8-a164-62fffef3d397';
+
+        if (!configLoaded) {
+            return;
+        }
+
+        if (!signalhubURL) {
+            return;
+        }
+
+        if (!stunServer) {
```

Comment:
can we show something to user if turn stun servers not configured. Or can we configure a default turn/stun serve./